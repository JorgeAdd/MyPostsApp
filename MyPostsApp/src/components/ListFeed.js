import React, { Component } from 'react';
import { View,Image, StyleSheet, TouchableOpacity } from 'react-native';
import { Card,CardItem, Body, Header, Container, Content, Fab, Icon, Footer, Button,Text, Input, Toast } from 'native-base';
import Display from 'react-native-display';
import AsyncStorage from '@react-native-community/async-storage';
import {FlatList} from 'react-native-gesture-handler';
export default class ListFeed extends Component {
  constructor(props) {
    super(props);
    this.state = {
      feedDisplay:true,
      newPostDisplay:false,
      feeds:"",
      titleNewPost:"",
      descNewPost:"",
      imageUriNewPost:require("../img/descarga.png"),
      locationNewPost:"Aguascalientesss Ags MX",
      postKey:0,
      orderBy:"date"
    };
  }

  componentDidMount(){
    //AsyncStorage.removeItem('posts')
    this.loadFeeds();
  }

  async loadFeeds(){
    var stringFeeds = await AsyncStorage.getItem('posts')
     this.setState({feeds:JSON.parse(stringFeeds)})
     console.log("this.state.feeds")
     console.log(this.state.feeds)
     this.setState({postKey:this.state.feeds.length})
  }

  addPostButton(){
    this.setState({feedDisplay:false,newPostDisplay:true})
  }

  async newPostMethod(){
    if(this.state.titleNewPost != ""){
      if(this.state.descNewPost != ""){
        this.setState({postKey: this.state.postKey+1})
    
        var newPost = 
        {
          key:""+this.state.postKey,
          title:""+this.state.titleNewPost,
          desc:""+this.state.descNewPost,
          img:this.state.imageUriNewPost,
          location:""+this.state.locationNewPost,
          datePosted: new Date()
        }
    
        var posts = await AsyncStorage.getItem('posts')
        if(posts != null){
          posts = JSON.parse(posts);
          posts.push(newPost)
        }else{
          posts = [newPost]
        }
        posts.sort(function(a,b){
          return new Date(b.datePosted) - new Date(a.datePosted)
        })
        await AsyncStorage.setItem('posts',JSON.stringify(posts))
        this.loadFeeds()
        this.setState({newPostDisplay:false,feedDisplay:true,titleNewPost:"",descNewPost:""})
      }else{
        Toast.show({
          text:"Description shouldn't be empty",
          position:'bottom',
          type:'danger',
          duration:5000
        })
      }
    }else{
      Toast.show({
        text:"Title shouldn't be empty",
        position:'bottom',
        type:'danger',
        duration:5000
      })
    }
  }

  cancelNewPost(){
    this.setState({feedDisplay:true,newPostDisplay:false,titleNewPost:"",descNewPost:""})
  }

  dateSort(){
    this.setState({orderBy:"date"})
    var posts = this.state.feeds
    posts.sort(function(a,b){
      return new Date(b.datePosted) - new Date(a.datePosted)
    })
    this.setState({feeds: posts})
  }

  titleSort(){
    this.setState({orderBy:"title"})
    var posts = this.state.feeds
    posts.sort(function(a,b){
      return a.title.localeCompare(b.title)
    })
    this.setState({feeds: posts})
    console.log(this.state.feeds);
    
  }

  render() {
    return (
      <Container>
        <View style={{flex:1}}>
          <Display enable={this.state.feedDisplay}>
            <FlatList data={this.state.feeds} renderItem={({item,index}) => (
            <Card>
              <Icon type="FontAwesome" name="trash" style={stylesListFeed.trashIcon} />
              <CardItem style={{ width: "100%" }}>
                <Image style={{ width: "100%" }} source={item.img} />
              </CardItem>
              <CardItem header>
                <Text style={stylesListFeed.headerPost}>{item.title}</Text>
              </CardItem>
              <CardItem style={{ flexDirection: "column" }}>
                <Body>
                  <Text style={stylesListFeed.descPost}>
                    {item.desc}
                </Text>
                  <Text style={stylesListFeed.datePost}>
                    {item.datePosted}
                </Text>
                </Body>
              </CardItem>
              <CardItem footer style={{ justifyContent: "center" }}>
                <Text style={stylesListFeed.footerPost}>
                  Posted in {item.location}. See in the map.
                </Text>
              </CardItem>
            </Card>
            )}
            key={item => item.key}
            keyExtractor={item => item.key.toString()}
            ListEmptyComponent={<Text style={{textAlign:"center",color:"gray"}}>You haven't posted anything yet!</Text>}
            ListFooterComponent={<Text>Pag 1</Text>}
            />
          </Display>

          <Display enable={this.state.newPostDisplay}>
            <Card>
              <CardItem header>
                <Text>Title *</Text>
              </CardItem>
              <CardItem>
                <Input onChangeText={title => this.setState({ titleNewPost: title })} style={stylesListFeed.inputFormat} />
              </CardItem>
              <CardItem>
                <Text>Description *</Text>
              </CardItem>
              <CardItem>
                <Input multiline={true} numberOfLines={20} onChangeText={desc => this.setState({ descNewPost: desc })} style={stylesListFeed.inputFormat} />
              </CardItem>
              <TouchableOpacity>
                <CardItem>
                  <Icon type="FontAwesome" name="image" /><Text> Upload an image</Text>
                </CardItem>
              </TouchableOpacity>
              <CardItem>
                <Text>Location</Text>
              </CardItem>
              <CardItem style={{justifyContent:"space-around"}}>
                <Button style={stylesListFeed.orderSelectedButton} onPress={()=>this.newPostMethod()}>
                  <Text style={stylesListFeed.orderSelectedText}>Post</Text>
                </Button>
                <Button style={stylesListFeed.orderUnselectedButton} onPress={()=>this.cancelNewPost()}>
                  <Text style={stylesListFeed.orderUnselectedText}>Cancel</Text>
                </Button>
              </CardItem>
            </Card>
          </Display>

        </View>

        {this.state.feedDisplay ? 
        <Footer style={{height:20,backgroundColor:"white",width:"100%", marginBottom:0,paddingBottom:0, flexDirection:"column",}}>
          <Text style={{marginTop:15,marginBottom:5}}>Order by:</Text>
          <View style={{flexDirection:"row",justifyContent:"space-around",width:"75%"}}>
            <Button style={this.state.orderBy == "date" ? stylesListFeed.orderSelectedButton : stylesListFeed.orderUnselectedButton}
             onPress={()=> this.dateSort()}>
              <Text style={this.state.orderBy == "date" ? stylesListFeed.orderSelectedText : stylesListFeed.orderUnselectedText}>
                Date</Text>
            </Button>
            <Button style={this.state.orderBy == "title" ? stylesListFeed.orderSelectedButton : stylesListFeed.orderUnselectedButton}
             onPress={()=> this.titleSort()}>
              <Text style={this.state.orderBy == "title" ? stylesListFeed.orderSelectedText : stylesListFeed.orderUnselectedText}>
                Title</Text>
            </Button>
          </View>
        <Fab
          active={this.state.feedDisplay}
          position="bottomRight"
          style={stylesListFeed.fabButton}
          onPress={() => this.addPostButton()}>
            <View style={stylesListFeed.fabView}>
            <Icon name="add" style={stylesListFeed.fabIcon}/>
            </View>
        </Fab>
        </Footer>
        :<></>}
        
      </Container>
    );
  }
}
const stylesListFeed = StyleSheet.create({
  trashIcon: {
    alignSelf:"center",
    color:"pink",
    marginTop:5,
    position:"absolute",
    zIndex:1,
    textAlign:"right",
    width:"95%"
  },
  headerPost: {
    fontSize:18,
    fontWeight:"bold"
  },
  descPost: {
    fontSize:16
  },
  datePost: {
    fontSize:12,
    color:"gray",
    textAlign:"left"
  },
  footerPost: {
    fontSize:12
  },
  orderSelectedButton: {
    width:"30%",
    justifyContent:"center",
    backgroundColor:"lightblue",
    borderWidth:1,
    borderColor:"blue"    
  },
  orderUnselectedButton: {
    width:"30%",
    justifyContent:"center",
    backgroundColor:"#FFF",
    borderWidth:1,
    borderColor:"blue"
  },
  orderSelectedText: {
    color:"blue"
  },
  orderUnselectedText: {
    color:"blue"
  },
  fabButton: {
    flex:1,
    backgroundColor:"lightblue",
    justifyContent:"center",
    alignItems:"center",
    alignContent:"center",
    alignSelf:"center"
  },
  fabView: { 
    width:"70%", 
    height:"70%",
    alignItems:"center"
  },
  fabIcon:{
    fontSize:40,
    color:"blue"
  },
  inputFormat: {
    borderColor:"lightblue",
    borderWidth:1,
    borderRadius:5,
    minHeight:50
  }
})
