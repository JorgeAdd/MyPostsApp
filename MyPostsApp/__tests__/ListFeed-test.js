import 'react-native';
import React from 'react';
import ListFeed from '../src/components/ListFeed';
import rerender from 'react-test-renderer';

test('ListFeed snapShot', () => {
  const snap = rerender.create(
      <ListFeed/>
  ).toJSON();
expect(snap).toMatchSnapshot();
})
