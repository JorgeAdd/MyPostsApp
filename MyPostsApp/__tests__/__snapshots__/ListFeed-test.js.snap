// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFeed snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      forwardedRef={[Function]}
      style={Object {}}
    >
      <RCTScrollView
        ListEmptyComponent={
          <Styled(Text)
            style={
              Object {
                "color": "gray",
                "textAlign": "center",
              }
            }
          >
            You haven't posted anything yet!
          </Styled(Text)>
        }
        ListFooterComponent={
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 10,
              }
            }
          >
            <React.Fragment />
            <React.Fragment />
            <React.Fragment />
          </View>
        }
        collapsable={false}
        data=""
        disableVirtualization={false}
        extraData={true}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        renderScrollComponent={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <Text
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "gray",
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
          >
            You haven't posted anything yet!
          </Text>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginVertical": 10,
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
        <View
          forwardedRef={[Function]}
          hideModalContentWhileAnimating={false}
          onModalHide={[Function]}
          onModalWillHide={[Function]}
          onModalWillShow={[Function]}
          pointerEvents="box-none"
          scrollHorizontal={false}
          scrollOffset={0}
          scrollOffsetMax={0}
          scrollTo={null}
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          swipeThreshold={100}
        >
          <AIRMap
            handlePanDrag={false}
            initialRegion={
              Object {
                "latitude": 0,
                "latitudeDelta": 0.0122,
                "longitude": 0,
                "longitudeDelta": 0.0121,
              }
            }
            onChange={[Function]}
            onLayout={[Function]}
            onMapReady={[Function]}
            onMarkerPress={[Function]}
            region={null}
            style={
              Object {
                "height": 200,
                "width": "100%",
              }
            }
          >
            <AIRMapMarker
              coordinate={
                Object {
                  "latitude": 0,
                  "longitude": 0,
                }
              }
              draggable={false}
              onPress={[Function]}
              stopPropagation={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "position": "absolute",
                  },
                  undefined,
                ]
              }
              title="Your post was submited in this location"
            />
          </AIRMap>
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                }
              }
              uppercase={false}
            >
              Ok
            </Text>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 20,
          "flexDirection": "column",
          "height": 56,
          "left": 20,
          "position": "absolute",
          "right": undefined,
          "top": undefined,
          "width": 56,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#fc5185",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 20,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "bottom": 50,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "left": 8,
            "marginBottom": 10,
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": null,
            "paddingTop": null,
            "position": "absolute",
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": undefined,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 40,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#fff",
                  "fontSize": 20,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingTop": 2,
                },
                Object {
                  "color": "#f5f5f5",
                  "fontSize": 25,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="FontAwesome"
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#fc5185",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 20,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "bottom": 50,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "left": 8,
            "marginBottom": 10,
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": null,
            "paddingTop": null,
            "position": "absolute",
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": undefined,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 40,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#fff",
                  "fontSize": 20,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingTop": 2,
                },
                Object {
                  "color": "#f5f5f5",
                  "fontSize": 25,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="FontAwesome"
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#3FC1C9",
            "borderRadius": 28,
            "bottom": 0,
            "elevation": 4,
            "flex": 1,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
            "width": 56,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "color": "#fff",
              "fontSize": 24,
              "height": "70%",
              "justifyContent": "center",
              "lineHeight": 27,
              "width": "70%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#364F6B",
                    "fontSize": 30,
                  },
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            type="FontAwesome"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 20,
          "flexDirection": "column",
          "height": 56,
          "left": undefined,
          "position": "absolute",
          "right": 20,
          "top": undefined,
          "width": 56,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#3FC1C9",
            "borderRadius": 28,
            "bottom": 0,
            "elevation": 4,
            "flex": 1,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
            "width": 56,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "color": "#fff",
              "fontSize": 24,
              "height": "70%",
              "justifyContent": "center",
              "lineHeight": 27,
              "width": "70%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#364F6B",
                    "fontSize": 40,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
